### DISCUSSION TITLE
Changing props with Controls re-renders the component

### DISCUSSION THREAD

**Discussion by spacecakes** [storybookjs/storybook#15131] [Help]
Using latest storybook with Vue and building a component that tweens the `value` prop it gets passed, meaning it will count up (or down) to the next value it gets passed, creating a nice little slot machine-style animation. 

I've been trying to figure out why my component doesn't actually animate when passed a new prop value, and the lifecycle hooks finally gave me the answer: the component is re-mounted whenever the value change, making the range slider I set up to demo it in Storybook useless. In other words, if I use the control to change the value, `mounted()` runs each time, meaning the value that gets passed is actually the _initial_ value. So, nothing to tween from.

I got around this issue by simply adding a `<button @click="...` in my template to transform the data for me, but it seems hacky and unnecessary. Am I doing something wrong or is this how Controls are set up to work?

**Comment by airtonix** [comment]
Then you've just exposed a flaw in how you've created your component. 👍🏻 

In react this is the difference between a controlled and uncontrolled component.

**Reply by spacecakes** [reply]
Nah it's controlled. I can't remember what the issue was though because it's no longer there after 18 months of updates to both it and Storybook. Problem solved I guess :D
